src/orion/core/worker/storage_backup.py

```diff
@@ -0,0 +1,473 @@
+                    )
+                    queries_to_delete.setdefault(COL_BENCHMARKS, []).append(
+                        {"_id": dst_benchmark["_id"]}
+                    )
+                elif resolve == "bump":
+                    raise DatabaseError(
+                        "Can't bump benchmark version, "
+                        "as benchmarks do not currently support versioning."
+                    )
+            # Delete benchmark database ID so that a new one will be generated on insertion
+            del src_benchmark["_id"]
+            data_to_add.setdefault(COL_BENCHMARKS, []).append(src_benchmark)
+        _describe_import_progress(
+            STEP_CHECK_BENCHMARKS,
+            len(src_benchmarks),
+            len(src_benchmarks),
+            progress_callback,
+        )
+
+    _describe_import_progress(STEP_CHECK_DST_EXPERIMENTS, 0, 1, progress_callback)
+    all_dst_experiments = dst_storage.fetch_experiments({})
+    # Dictionary mapping dst exp name to exp version to list of exps with same name and version
+    dst_exp_map = {}
+    last_experiment_id = max((data["_id"] for data in all_dst_experiments), default=0)
+    last_versions = {}
+    for dst_exp in all_dst_experiments:
+        name = dst_exp["name"]
+        version = dst_exp["version"]
+        last_versions[name] = max(last_versions.get(name, 0), version)
+        dst_exp_map.setdefault(name, {}).setdefault(version, []).append(dst_exp)
+    _describe_import_progress(STEP_CHECK_DST_EXPERIMENTS, 1, 1, progress_callback)
+
+    for i, experiment in enumerate(experiments):
+        _describe_import_progress(
+            STEP_CHECK_SRC_EXPERIMENTS, i, len(experiments), progress_callback
+        )
+        dst_experiments = dst_exp_map.get(experiment["name"], {}).get(
+            experiment["version"], []
+        )
+        if dst_experiments:
+            (dst_experiment,) = dst_experiments
+            if resolve == "ignore":
+                logger.info(
+                    f"Ignored experiment already in dst: "
+                    f'{experiment["name"]}.{experiment["version"]}'
+                )
+                continue
+            if resolve == "overwrite":
+                # We must remove experiment data in dst
+                logger.info(
+                    f"Overwrite experiment in dst: "
+                    f'{dst_experiment["name"]}.{dst_experiment["version"]}'
+                )
+                for collection in EXPERIMENT_RELATED_COLLECTIONS:
+                    queries_to_delete.setdefault(collection, []).append(
+                        {"experiment": dst_experiment["_id"]}
+                    )
+                queries_to_delete.setdefault(COL_EXPERIMENTS, []).append(
+                    {"_id": dst_experiment["_id"]}
+                )
+            elif resolve == "bump":
+                old_version = experiment["version"]
+                new_version = last_versions.get(experiment["name"], 0) + 1
```

Comment:
This will not work for all storage backends. The new ID should be inferred after calling `storage.create_experiment` as it is the storage backend that will generate it. That unfortunately means that we will need deferring the trial.experiment assignment to `_execute_import`.

Reply:
Oh, that's for a version, not an ID.